pkg/controller/nstemplatetier/nstemplatetier_controller.go

```diff
@@ -27,100 +29,164 @@ import (
-	logger.Info("Reconciling NSTemplateTier")
 
-	// Fetch the NSTemplateTier tier
+	// fetch the NSTemplateTier tier
 	tier := &toolchainv1alpha1.NSTemplateTier{}
-	err := r.client.Get(context.TODO(), request.NamespacedName, tier)
-	if err != nil {
+	if err := r.client.Get(context.TODO(), request.NamespacedName, tier); err != nil {
 		if errors.IsNotFound(err) {
-			log.Info("NSTemplateTier not found")
+			logger.Info("NSTemplateTier not found")
 			return reconcile.Result{}, nil
 		}
 		// Error reading the object - requeue the request.
 		logger.Error(err, "unable to get the current NSTemplateTier tier")
 		return reconcile.Result{}, err
 	}
-	if err := r.ensureTemplateUpdateRequest(logger, tier); err != nil {
+
+	// create a new entry in the `status.history`
+	if added, err := r.ensureStatusUpdateRecord(logger, tier); err != nil {
+		logger.Error(err, "unable to insert a new entry in status.updates after NSTemplateTier changed.")
+		return reconcile.Result{}, err
+	} else if added {
+		logger.Info("Requeing after adding a new entry in tier.status.updates")
+		return reconcile.Result{Requeue: true}, nil
+	}
+	if done, err := r.ensureTemplateUpdateRequest(logger, tier); err != nil {
 		logger.Error(err, "unable to ensure TemplateRequestUpdate resource after NSTemplateTier changed.")
 		return reconcile.Result{}, err
+	} else if done {
+		log.Info("No MasterUserRecord to update. Setting the completion timestamp immediately")
+		if err := r.markUpdateRecordAsCompleted(tier); err != nil {
+			logger.Error(err, "unable to mark latest status.update as complete.")
+			return reconcile.Result{}, err
+		}
+		logger.Info("No MasterUserRecord to update")
+		return reconcile.Result{}, nil
 	}
 	return reconcile.Result{}, nil
 }
 
+// ensureStatusUpdateRecord adds a new entry in the `status.updates` with the current date/time
+// if needed and cleans-up the previous one if needed (ie, clears the `failedAccounts` array)
+// returns `true` if an entry was added, `err` if something wrong happened
+func (r *NSTemplateTierReconciler) ensureStatusUpdateRecord(logger logr.Logger, tier *toolchainv1alpha1.NSTemplateTier) (bool, error) {
+	hash, err := ComputeHashForNSTemplateTier(*tier)
+	if err != nil {
+		return false, errs.Wrapf(err, "unable to append an entry in the `status.updates` for NSTemplateTier '%s'", tier.Name)
+	}
+	// if there was no previous status:
+	if len(tier.Status.Updates) == 0 {
+		tier.Status.Updates = append(tier.Status.Updates, toolchainv1alpha1.NSTemplateTierHistory{
+			StartTime: metav1.NewTime(time.Now()),
+			Hash:      hash,
+		})
+		return true, r.client.Status().Update(context.TODO(), tier)
+	}
+	// check that last entry
+	if tier.Status.Updates[len(tier.Status.Updates)-1].Hash == hash {
+		logger.Info("current update still in progress")
+		return false, nil
+	}
+	tier.Status.Updates[len(tier.Status.Updates)-1].FailedAccounts = nil // reset
```

Comment:
Why are we resetting the previous failed account count?

Reply:
that's something @MatousJobanek suggested when we discussed about storing the usernames for whom the update failed: these usernames are only worth keeping for the current update, so when we start a new one, we should reset the previous `FailedAccounts`, in order to avoid polluting the `Status.Updates` with too much data. 

I've clarified the comment in f9060c8

Reply:
> I've clarified the comment in f9060c8

Did you forget to commit this? I don't see f9060c8 commit here.

Reply:
indeed, I forgot about to push it (I thought I did, though 🤨)

Reply:
OK. It does make sense. And thanks for adding the comment. It helps.